bun/lib/dependabot/bun/update_checker/version_resolver.rb

```diff
@@ -229,48 +309,71 @@ def updated_monorepo_dependencies
           updates
         end
 
+        sig { returns(T.nilable(Dependabot::Dependency)) }
         def types_package
-          @types_package ||= begin
+          return @types_package if @types_package
+
+          @types_package = begin
             types_package_name = PackageName.new(dependency.name).types_package_name
             top_level_dependencies.find { |d| types_package_name.to_s == d.name } if types_package_name
           end
+          @types_package
         end
 
+        sig { returns(T.nilable(Dependabot::Dependency)) }
         def original_package
-          @original_package ||= begin
+          return @original_package if @original_package
+
+          @original_package = begin
             original_package_name = PackageName.new(dependency.name).library_name
             top_level_dependencies.find { |d| original_package_name.to_s == d.name } if original_package_name
           end
+          @original_package
         end
 
+        sig { returns(T.nilable(Dependabot::Version)) }
         def latest_types_package_version
-          @latest_types_package_version ||= latest_version_finder(types_package).latest_version_from_registry
+          types_pkg = types_package
+          return unless types_pkg
+
+          return @latest_types_package_version if @latest_types_package_version
+
+          @latest_types_package_version = latest_version_finder(types_pkg).latest_version_from_registry
+          @latest_types_package_version
         end
 
+        sig { returns(T::Boolean) }
         def types_update_available?
-          return false if types_package.nil?
+          types_pkg = types_package
+          return false unless types_pkg
 
-          return false if latest_types_package_version.nil?
+          latest_types_version = latest_types_package_version
+          return false unless latest_types_version
```

Comment:
nit, introducing a similarly named variable makes this confusing

```suggestion
          return false if latest_types_package_version.nil?
```

Reply:
I need to create variable to satisfy sorbet for checks. That's why I am using a variable and returning. Otherwise on down, sorbet will not satisfy the check and still will ask for typings.